### PR TITLE
feat: make <Dropzone> headless

### PR DESCRIPTION
**Pull request by rolandjitsu**
**What kind of change does this PR introduce?**

- [ ] bugfix
- [x] feature
- [x] refactoring / style
- [ ] build / chore
- [x] documentation

**Did you add tests for your changes?**

- [x] Yes, my code is well tested
- [ ] Not relevant

**If relevant, did you update the documentation?**

- [x] Yes, I've updated the documentation
- [ ] Not relevant

**Summary**

This PR is a continuation of https://github.com/react-dropzone/react-dropzone/pull/468 that includes the latest changes made since then.

@okonet it was easier for me to start working on a new PR than continue in your branch. I hope you don't mind.

I will progressively update the docs and examples until it reflects the new API.

Things left todo:
- [x] Update docs
- [x] Update examples
- [x] Improve accessibility (focus and tab)
- [x] Update typings

**Does this PR introduce a breaking change?**

Yes, this PR removes all style management and makes the component headless. meaning users need to provide a render fn (or a children fn).

Today:
```jsx
<Dropzone>
    <span>Drop files here</span>
<Dropzone>
```

After merging:
```jsx
<Dropzone>
  {({ getRootProps, getInputProps }) => (
    <div {...getRootProps()}>
      <input {...getInputProps()} />
    </div>
  )}
</Dropzone>
```

**Other information**

### CODE REVIEW

**Comment by okonet** [comment]
Is this based on https://github.com/react-dropzone/react-dropzone/pull/468?

**Comment by okonet** [comment]
Nice! So you just squashed all commits into one?

**Comment by okonet** [comment]
Let me know when it’s ready for review! Looking great so far.

**Comment by okonet** [comment]
I’m still not convinced about the need of `getRoptProps` and `getInputProps` tbh but it’s your call now

**Comment by okonet** [comment]
My proposal and initial direction was to make the dropzone inject input automatically into the root component. So basically Dropzone is the `input` and takes all of its props plus other props that are required for it to function. But I see how this also can be a bit ambiguous.

**Comment by rolandjitsu** [comment]
@okonet I understand. But I think this change gives users a lot more flexibility. Let's see how users react to this change and if it turns out it's not received very well we'll think of a different way to handle this.

File: README.md [hunk 1]
```diff
@@ -33,36 +33,96 @@ yarn add react-dropzone
...
+```jsx static
+<Dropzone>
+  {({getRootProps}) => <div {...getRootProps()} />}
+</Dropzone>
+```
+
+Or as a prop of the dropzone:
+```jsx static
+<Dropzone
+  render={({getRootProps}) => <div {...getRootProps()} />}
```
Comment (okonet) [inline README.md]: Why support `render` even?

File: examples/Accept/Readme.md [hunk 2]
```diff
@@ -59,37 +64,35 @@ Because of HTML5 File API differences across different browsers during the drag,
...
-    }
-    return "Dropping some files here...";
-  }}
+```jsx harmony
+<Dropzone accept=".jpeg,.png">
+  {({ getRootProps, getInputProps, isDragActive, isDragAccept, isDragReject }) => (
+    <div {...getRootProps()} className="dropzone">
+      <input {...getInputProps()} />
+      <div>
+        {isDragAccept && "All files will be accepted"}
```
Comment (okonet) [inline examples/Accept/Readme.md]: Isn't this one going to render `false` if `isDragAccept` is false?

**Comment by okonet** [comment]
Could you please include #612 ?

File: README.md (hunk 1 above)
Comment (rolandjitsu) [reply README.md]: Good point, it was in the prev PR, didn't think to remove it.

**Comment by rolandjitsu** [comment]
@okonet please see the `/examples/Styling/Readme.md` file in this PR, I have included the examples for styling with inline styles or `styled-components`

File: examples/Accept/Readme.md (hunk 2 above)
Comment (rolandjitsu) [reply examples/Accept/Readme.md]: Nope, it seems to not render `false`.

**Comment by chasestarr** [comment]
hey folks - could you use any help here? this feature would be great for my project

**Comment by okonet** [comment]
Congratulations!

**Pull request merged**